test/as-json.js

```diff
@@ -70,29 +70,32 @@ test('getting an ok response', function t(assert) {
             '-J'
         ];
 
-        tcurl.exec(cmd, onResponse);
-
-        function onResponse(err, resp) {
-            assert.ifError(err);
-            assert.deepEqual(resp, {
-                ok: true,
-                head: null,
-                body: {
-                    opts: {
-                        isOptions: true
+        tcurl.exec(cmd, {
+            error: function error(err) {
+                assert.ifError(err);
```

Comment:
assert.ifError can fail the test. I am actually OK either way. However, we need to keep consistent about using "assert.ifError" v.s. "if (err) { ... }" (actually suggested by you :)). Maybe, there is anything special here I am missing?